### PR TITLE
vendor: update govmm package

### PR DESCRIPTION
**Pull request by liubin**
New govmm package deleted `-realtime` option, so this
commit also delete the `RealTime` property from the hypervisor.

### CODE REVIEW

**Comment by Jakob-Naucke** [comment]
@liubin thank you for your PR. Wouldn't this be a new API version, e.g. 1.0 -> 1.1?

**Comment by fidencio** [comment]
@liubin, as stable-2.3 is using QEMU 6.1, wouldn't make sense to also backport this one?  Otherwise the breakage that you're trying to avoid with https://github.com/kata-containers/govmm/issues/205 will be seen on stable-2.3.

**Comment by liubin** [comment]
> @liubin, as stable-2.3 is using QEMU 6.1, wouldn't make sense to also backport this one? Otherwise the breakage that you're trying to avoid with [kata-containers/govmm#205](https://github.com/kata-containers/govmm/issues/205) will be seen on stable-2.3.

Ok, I will backport it to stable-2.3

**Comment by liubin** [comment]
> @liubin thank you for your PR. Wouldn't this be a new API version, e.g. 1.0 -> 1.1?

We do not strictly maintain the version number of the Virtcontainer API, the API  version has been 1.0 for a long time.

**Review by Jakob-Naucke — APPROVED** [review]

**Review by lifupan — APPROVED** [review]

**Comment by bergwolf** [comment]
Thanks for the PR! And I would like to ask why not just make a copy of it and move it inside kata-containers repo, so that at least we don't have to bother to update two places every time, which is the main complaint about govmm in the past years. I see that @dgibson has abandoned https://github.com/kata-containers/kata-containers/issues/2393. Can we revive that one? It is a pure win for us to just keep the code piece in the same repo.

**Comment by Jakob-Naucke** [comment]
See also http://lists.katacontainers.io/pipermail/kata-dev/2021-August/002101.html

**Comment by liubin** [comment]
@Jakob-Naucke Is the new govmm package not compatible with the QEMU 5.2.0 ?

**Comment by Jakob-Naucke** [comment]
@liubin Kata seems incompatible with QEMU 5.2.0 regardless of GoVMM update: `qemu-system-x86_64: -object memory-backend-file,id=mem0,mem-path=/usr/share/kata-containers/kata-containers.img,size=134217728,readonly=on: Property 'memory-backend-file.readonly' not found`

**Comment by Jakob-Naucke** [comment]
The difference I can find in the QEMU invocations from the logs is `-overcommit mem-lock=on` (from https://github.com/kata-containers/govmm/pull/190), but that seems to work for me...

**Comment by liubin** [comment]
I'm not clear that why QEMU 5.2.0 is used, I can't find it in `versions.yaml`

**Comment by Jakob-Naucke** [comment]
@liubin I don't think QEMU 5.2.0 is being used at all, what makes you think so?

**Comment by liubin** [comment]
> @liubin I don't think QEMU 5.2.0 is being used at all, what makes you think so?

I'm sorry, the 5.2.0 is `qemu-img` package, QEMU 6.1.0 is used.

File: src/runtime/go.mod [hunk 1]
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/hashicorp/go-multierror v1.0.0
 	github.com/intel-go/cpuid v0.0.0-20210602155658-5747e5cec0d9
-	github.com/kata-containers/govmm v0.0.0-20210909155007-1b60b536f3c7
+	github.com/kata-containers/govmm v0.0.0-20211118155643-d432e21d6fc0
```
Comment (devimc) [inline src/runtime/go.mod:30]: can we use the latest version of govmm `aaac742` ?

**Comment by jodh-intel** [comment]
Any update on this @liubin?

**Comment by liubin** [comment]
> Any update on this @liubin?

I will keep it and wait until all CIs pass. Now some CI always fails, it's may be some issues there.

**Comment by jodh-intel** [comment]
Any update on this @liubin?

**Comment by dgibson** [comment]
> Thanks for the PR! And I would like to ask why not just make a copy of it and move it inside kata-containers repo, so that at least we don't have to bother to update two places every time, which is the main complaint about govmm in the past years. I see that @dgibson has abandoned #2393. Can we revive that one? It is a pure win for us to just keep the code piece in the same repo.

I agree that having these together would be a good idea.  The difficulty is in balancing Kata's needs against the (few, but extant) external users of govmm.  I simply haven't had time to sort this out, and am unlikely to any time soon.

File: src/runtime/go.mod (hunk 1 above)
Comment (liubin) [reply src/runtime/go.mod]: Updated to `aaac742`

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

**Comment by liubin** [comment]
Too many CI jobs failed......

**Comment by devimc** [comment]
@liubin  I have identified the bug, I will fix it and update govmm in this PR https://github.com/kata-containers/kata-containers/pull/3460

**Pull request closed without merging**